src/modules/fancyzones/lib/FancyZones.cpp

```diff
@@ -404,6 +410,79 @@ void FancyZones::MoveWindowIntoZone(HWND window, winrt::com_ptr<IZoneWindow> zon
     }
 }
 
+int RectWidth(const RECT& rect)
+{
+    return rect.right - rect.left;
+}
+
+int RectHeight(const RECT& rect)
+{
+    return rect.bottom - rect.top;
+}
+
+RECT FitOnScreen(HWND window, const RECT& windowRect, HMONITOR monitor, const RECT& monitorRect) 
+{
+    int left = monitorRect.left + windowRect.left;
+    int top  = monitorRect.top  + windowRect.top;

```

Comment:
The monitor on which `window` was spawned may not have its (left, top) equal to (0, 0), I think we need to subtract its (left, top) here.

Reply:
You're right, if the taskbar is on the top or on the left that can change.

Reply:
Or more importantly, when the primary monitor is in the middle (3 monitors)

Reply:
Primary monitor always has screen left-top coordinate in (0,0). Work area might not be at (0,0) because of the taskbar. If the primary monitor is in the middle than monitor on the left has negative coordinates.

Reply:
Taken (left, top) from origin monitor into consideration.